### PR TITLE
feat: login route added

### PR DESCRIPTION
**Pull request by Arnab-31**
#21

### CODE REVIEW

File: server/src/models/user.js [hunk 1]
```diff
@@ -26,4 +28,54 @@ const userSchema = new mongoose.Schema({
...
+
+  delete userObject.password;
+  delete userObject.tokens;
+
+  return userObject;
+}
+
+// creates and saves a jwt token to the user record and returns the token
+userSchema.methods.generateAuthToken = async function() {                     
+  const token = jwt.sign({_id: this._id.toString() }, process.env.JWT_SECRET, {
```
Comment (BhavyaaArora-08) [inline server/src/models/user.js]: dont do id.toString(), directly ObjectId hi pass krdo.

File: server/src/routers/api/userRouter.js [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
+    try{
+        await user.save();
+        const token = await user.generateAuthToken();
+        const filtered_user = user.toJSON();
+        res.status(201).json({msg: "You have been logged in successfully",user:filtered_user, token});
+    }catch(e){
+        res.status(400).send({ errors: [{msg: "Unable to login"}] });
+    } 
+})
+
```
Comment (BhavyaaArora-08) [inline server/src/routers/api/userRouter.js:35]: No, this is not what we are supposed to do.
Why create a new user when they are logging in.

Search if this user exists first, then alot a token

Use the findByCredentials function you created earlier.

**Comment by Arnab-31** [comment]
I have made the changes

**Comment by BhavyaaArora-08** [comment]
When a user logs in
1. Findbycredentials 
2. If no such user exists throw an error
3. Else assign a token and return according to the response format.

Once you are done, please test the API routes and send a screenshot here as well.
You can use Postman to test API

**Comment by BhavyaaArora-08** [comment]
Have you made the changes? @Arnab-31

**Comment by Arnab-31** [comment]
image
image
image
image

**Comment by Arnab-31** [comment]
@Codess08 I have added both the register and login routes along with proper error handling

**Pull request closed without merging**